.github/workflows/proto-registry.yml

```diff
@@ -0,0 +1,92 @@
+name: Buf Push
+# Protobuf runs buf (https://buf.build/) push updated proto files to https://buf.build/provenance-io
+# This workflow is only run when a .proto file has been changed or when this workflow has been modifed.
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "**/**.proto"
+      - ".github/workflows/proto-registry.yml"
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"     # Push events to matching v*, i.e. v1.0, v20.15.10
+      - "v[0-9]+.[0-9]+.[0-9]+-rc*" # Push events to matching v*, i.e. v1.0-rc1, v20.15.10-rc5
+
+# Concurrency is not defined in here because this job usually doesn't run
+# long enough to need cancelling, and it's okay for it to run multiple times.
+
+# Provenance publishes Protobuf content (including third_party dependencies) to buf.build.
+# Because of third_party dependencies declared in buf.yml, a buf.lock will be generated
+# containing the dependency manifest. Among other things, the buf.lock contains the
+# latest commit hash of the third_party library in the buf.build Provenance repository.
+# When new files are added to the third_party library we need:
+#   1. publish the third_party library
+#   2. update buf.lock to point to the new commit hash
+#   3. sign and create a PR to commit buf.lock changes back to main
+# This helps avoid breaking changes when we cut new releases.
```

Comment:
It's not helping avoid breaking changes. It's helping avoid a buf push failure due to a new third party proto reference.

Reply:
updated.